metasyn/metaframe.py

```diff
@@ -120,46 +122,70 @@ def fit_dataframe(
         MetaFrame:
             Initialized metasyn metaframe.
         """
-        if privacy is None:
-            privacy = BasicPrivacy()
-        if spec is None:
-            spec = {}
+        if meta_config is None:
+            if privacy is None:
+                privacy = {"name": "none"}
+            elif isinstance(privacy, BasePrivacy):
+                privacy = privacy.to_dict()
+            var_specs = [] if var_specs is None else var_specs
+            dist_providers = dist_providers if dist_providers is not None else ["builtin"]
+            meta_config = MetaConfig(var_specs, dist_providers, privacy)
         else:
-            spec = deepcopy(spec)
+            assert privacy is None
 
-        if set(list(spec)) - set(df.columns):
-            raise ValueError(
-                "Argument 'spec' includes the specifications for column names that do "
-                "not exist in the supplied dataframe:"
-                f" '{set(list(spec)) - set(df.columns)}'")
+        if isinstance(df, pd.DataFrame):
+            df = pl.DataFrame(df)
```

Comment:
Maybe change this from `isinstance(df, pd.DataFrame)` to `not isinstance(df, pl.DataFrame)` and then try the conversion. If there is a good error message this will automatically check everything and also support everything that polars supports!

Reply:
Makes sense!